tests/integration/test_ethernet.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+import pytest
+import subprocess
+import sys
+
+try:
+    from unittest import mock
+except ImportError:
+    import mock
+
+
+def _configure_network(connections):
+    parentdir = os.path.normpath(os.path.join(os.path.dirname(__file__), "../../"))
+    with mock.patch.object(
+        sys,
+        "path",
+        [parentdir, os.path.join(parentdir, "module_utils/network_lsr")] + sys.path,
+    ):
+        with mock.patch.dict(
+            "sys.modules",
+            {
+                "ansible": mock.Mock(),
+                "ansible.module_utils": __import__("module_utils"),
+            },
+        ):
+            import library.network_connections as nc
+
+    class TestRunEnvironment(nc.RunEnvironment):
+        def log(self, connections, idx, severity, msg, **kwargs):
+            if severity == nc.LogLevel.ERROR:
+                print("Error: {}".format(connections[idx]))
+                raise RuntimeError(msg)
+            else:
+                print("Log: {}".format(connections[idx]))
+
+        def _check_mode_changed(self, *args, **kwargs):
+            pass
+
+    run_env = TestRunEnvironment()
+
+    cmd = nc.Cmd.create(
+        "nm",
+        run_env=run_env,
+        connections_unvalidated=connections,
+        connection_validator=nc.ArgValidator_ListConnections(),
+    )
+
+    cmd.run()
+
+
+@pytest.fixture
+def testnic1():
+    veth_name = "testeth"
+    try:
+        subprocess.run(
+            [
+                "ip",
+                "link",
+                "add",
+                veth_name,
+                "type",
+                "veth",
+                "peer",
+                "name",
+                veth_name + "peer",
+            ]
+        )
+        yield veth_name
+    finally:
+        subprocess.run(["ip", "link", "delete", veth_name])
+
+
+def _get_ip_addresses(connection_name):
+    ip_a = subprocess.run(
+        ["ip", "address", "show", connection_name], capture_output=True
+    )
+    interface = ip_a.stdout.decode("UTF-8")
+    return interface
```

Comment:
```suggestion
    return ip_a.stdout.decode("UTF-8")
```

`interface` does not seem to be a proper name but it also seems the variable is not needed.

Reply:
True! Fixed